py/tabcheck.py

```diff
@@ -0,0 +1,46 @@
+#!/usr/bin/python3 -B
+# © 2021 and later: Unicode, Inc. and others.
+# License & terms of use: http://www.unicode.org/copyright.html
+
+# This script checks .java source files for the presence of tab characters. If a tab is encountered,
+# the script will print the name (path) of the file. If any file(s) had tabs, the script will exit
+# with a non-zero status code (unless the --exit-zero option is specified, which forces a
+# zero/success status code return).
+
+import argparse
+from pathlib import Path
+import sys
+
+def main() -> bool:
+    ap = argparse.ArgumentParser()
+    ap.add_argument("ROOT_DIR",
+                    nargs='?',
+                    default=Path.cwd(),
+                    help="Top-level directory which will be recursively searched "
+                         "for .java files to be checked.")
+    ap.add_argument('--exit-zero',
+                    action='store_true',
+                    help="Force exit with status code 0 even if there are errors.")
+
+    args = ap.parse_args()
+    root = Path(args.ROOT_DIR)
+
+    if not root.exists():
+        print(f'{root.absolute()}: directory not found; aborting.')
+        return False
+
+    error_count = 0
```

Comment:
Why do you count the errors when you only return a boolean?

Reply:
Ah, good point. I was originally going to return the error_count (for a non-zero exit code) but changed my mind later on without modifying that bit. I can change that to "errors_found" and treat it as a boolean if you prefer. I don't think it makes much difference.

Reply:
I would like that. I can quickly re-approve.

Reply:
done in https://github.com/unicode-org/unicodetools/pull/207/commits/af7a0e13c7d04b1a16157936971ab2348b1e047a